From 00e462c63eeafdb75b532cdd58b5e3dd6a480224 Mon Sep 17 00:00:00 2001
From: dtodoroff <d.todorov@vekatech.com>
Date: Sun, 3 Nov 2024 11:14:34 +0200
Subject: [PATCH 7/9] add support for 6 vkboards in build system


diff --git a/arch/arm/dts/Makefile b/arch/arm/dts/Makefile
index e79531c283..59df885926 100644
--- a/arch/arm/dts/Makefile
+++ b/arch/arm/dts/Makefile
@@ -933,7 +933,13 @@ dtb-$(CONFIG_RCAR_GEN3) += \
 	rzg2ul-type2-dev.dtb \
 	smarc-rzg2ul.dtb \
 	rzg3s-dev.dtb \
-	smarc-rzg3s.dtb
+	smarc-rzg3s.dtb \
+        vkrzg2l.dtb \
+	vkrzg2lc.dtb \
+	vkcmg2lc.dtb \
+	vkrzv2l.dtb \
+	vkrzg2ul.dtb \
+	vk-d184280e.dtb
 
 ifdef CONFIG_RCAR_GEN3
 DTC_FLAGS += -R 4 -p 0x1000
diff --git a/arch/arm/mach-rmobile/Kconfig.64 b/arch/arm/mach-rmobile/Kconfig.64
index 28b10473c4..073b6d41e7 100644
--- a/arch/arm/mach-rmobile/Kconfig.64
+++ b/arch/arm/mach-rmobile/Kconfig.64
@@ -200,6 +200,36 @@ config TARGET_RZG2LC_DEV
 	help
           Support for Renesas RZ/G2LC Dev Platform
 
+config TARGET_VKRZG2L
+        bool "Vekatech VK-RZ/G2L Board "
+        help
+          Support for Vekatech VK-RZ/G2L Board
+
+config TARGET_VKRZG2LC
+        bool "Vekatech VK-RZ/G2LC Board "
+        help
+          Support for Vekatech VK-RZ/G2LC Board
+
+config TARGET_VKCMG2LC
+        bool "Vekatech VK-CMRZ/G2LC Board "
+        help
+          Support for Vekatech VK-CMRZ/G2LC Board
+
+config TARGET_VKRZV2L
+        bool "Vekatech VK-RZ/V2L Board "
+        help
+          Support for Vekatech VK-RZ/V2L Board
+
+config TARGET_VKRZG2UL
+        bool "Vekatech VK-RZ/G2UL Board "
+        help
+          Support for Vekatech VK-RZ/G2UL Board
+
+config TARGET_VK_D184280E
+        bool "Vekatech VK-D184280E Board "
+        help
+          Support for Vekatech VK-D184280E Board
+
 config TARGET_SMARC_RZG2LC
 	bool "RZ/G2LC SMARC board"
 	help
@@ -255,6 +285,12 @@ source "board/renesas/rzv2l-dev/Kconfig"
 source "board/renesas/rzg2lc-dev/Kconfig"
 source "board/renesas/rzg2ul-dev/Kconfig"
 source "board/renesas/rzg3s-dev/Kconfig"
+source "board/renesas/vkrzg2l/Kconfig"
+source "board/renesas/vkrzg2lc/Kconfig"
+source "board/renesas/vkcmg2lc/Kconfig"
+source "board/renesas/vkrzv2l/Kconfig"
+source "board/renesas/vkrzg2ul/Kconfig"
+source "board/renesas/vk-d184280e/Kconfig"
 
 config MULTI_DTB_FIT_UNCOMPRESS_SZ
 	default 0x80000 if TARGET_HIHOPE_RZG2
-- 
2.34.1

